link/uprobe.go

```diff
@@ -67,42 +67,66 @@ func OpenExecutable(path string) (*Executable, error) {
 		return nil, fmt.Errorf("parse ELF file: %w", err)
 	}
 
-	var ex = Executable{
+	ex := Executable{
 		path:    path,
-		symbols: make(map[string]elf.Symbol),
+		offsets: make(map[string]uint64),
 	}
-	if err := ex.addSymbols(se.Symbols); err != nil {
+
+	if err := ex.load(se, se.Symbols); err != nil {
 		return nil, err
 	}
 
-	if err := ex.addSymbols(se.DynamicSymbols); err != nil {
+	if err := ex.load(se, se.DynamicSymbols); err != nil {
 		return nil, err
 	}
 
 	return &ex, nil
 }
 
-func (ex *Executable) addSymbols(f func() ([]elf.Symbol, error)) error {
-	// elf.Symbols and elf.DynamicSymbols return ErrNoSymbols if the section is not found.
-	syms, err := f()
+func (ex *Executable) load(f *internal.SafeELFFile, fun func() ([]elf.Symbol, error)) error {
```

Comment:
Nit: we should just get rid of `fun` and do the calls to Symbols and DynamicSymbols from this function. Then concatenate one slice to the other and do the loop below.